pipelines/templates/install-dependencies.yml

```diff
@@ -0,0 +1,56 @@
+parameters:
+- name: platform
+  type: string
+
+steps:
+- template: config-version.yml
+  parameters:
+    platform: ${{ parameters.platform }}
+
+- script: |
+    set -e
+    sudo apt-get install -y swig3.0
+    sudo rm /usr/bin/swig
+    sudo ln -s swig3.0 /usr/bin/swig
+  displayName: (Ubuntu) Downgrade swig
+  condition: and(succeeded(), contains('${{ parameters.platform }}', 'ubuntu'))
+
+- script: |
+    set -e
+    brew install 'swig@3'
+    rm /usr/local/bin/swig
+    ln -s '/usr/local/opt/swig@3/bin/swig' /usr/local/bin/swig
+  displayName: (macOS) Downgrade swig
+  condition: and(succeeded(), contains('${{ parameters.platform }}', 'macos'))
+
+- task: UniversalPackages@0
+  inputs:
+    vstsFeed: NNIOpenSource/dependencies
```

Comment:
I think if a PR tries to upgrade ubuntu-latest to, let's say, Python 3.10. When it runs the pipeline check, it will get the wrong binary, which causes pipeline failure.

But on the other hand, this should not happen often. And since dependencies expire every day, that should not take long to heal itself.

Reply:
Upgrading to new minor version is save, because the prefix will change and the cache will do nothing.

I don't have confidence about micro/patch version though.
There was a micro version upgrade during the development and caused a problem.
I have fixed that specific problem but I don't know if there are more.